PR: https://github.com/odoo/odoo/pull/

From: b5780cf8bb07f98509d854f72d2c7e27585bdbae
From: Arnold Moyaux
Date: 2019-04-12 12:50:34

Structural Changes: 5
Total Changes: 6

[FIX] mrp: by-products typo + default type on product

Be consistent with by-products and byproducts. Only use by-products.
When a product is created from the BoM view, it has a default type
as storable product, set the same type when the product is created
from by-products tab.

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -364,13 +364,13 @@ class MrpSubProduct(models.Model):
     _name = 'mrp.subproduct'
     _description = 'Byproduct'
 
-    product_id = fields.Many2one('product.product', 'Product', required=True)
+    product_id = fields.Many2one('product.product', 'By-product', required=True)
     product_qty = fields.Float(
-        'Product Qty',
+        'Quantity',
         default=1.0, digits=dp.get_precision('Product Unit of Measure'), required=True)
     product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True)
     bom_id = fields.Many2one('mrp.bom', 'BoM', ondelete='cascade')
-    operation_id = fields.Many2one('mrp.routing.workcenter', 'Produced at Operation')
+    operation_id = fields.Many2one('mrp.routing.workcenter', 'Produced in Operation')
 
     @api.onchange('product_id')
     def onchange_product_id(self):
